Tests/FeedStoreChallengeTests.swift

```diff
@@ -107,50 +168,70 @@ class FeedStoreChallengeTests: XCTestCase, FeedStoreSpecs {
 //  ***********************
 
-//extension FeedStoreChallengeTests: FailableRetrieveFeedStoreSpecs {
-//
-//	func test_retrieve_deliversFailureOnRetrievalError() throws {
-////		let sut = try makeSUT()
-////
-////		assertThatRetrieveDeliversFailureOnRetrievalError(on: sut)
-//	}
-//
-//	func test_retrieve_hasNoSideEffectsOnFailure() throws {
-////		let sut = try makeSUT()
-////
-////		assertThatRetrieveHasNoSideEffectsOnFailure(on: sut)
-//	}
-//
-//}
+extension FeedStoreChallengeTests: FailableRetrieveFeedStoreSpecs {
 
-//extension FeedStoreChallengeTests: FailableInsertFeedStoreSpecs {
-//
-//	func test_insert_deliversErrorOnInsertionError() throws {
-////		let sut = try makeSUT()
-////
-////		assertThatInsertDeliversErrorOnInsertionError(on: sut)
-//	}
-//
-//	func test_insert_hasNoSideEffectsOnInsertionError() throws {
-////		let sut = try makeSUT()
-////
-////		assertThatInsertHasNoSideEffectsOnInsertionError(on: sut)
-//	}
-//
-//}
+	func test_retrieve_deliversFailureOnRetrievalError() throws {
+		let sut = try makeSUT()
+		
+		activateNSManagedObjectContextRetrievalFailure()
+		
+		assertThatRetrieveDeliversFailureOnRetrievalError(on: sut)
+		
+		deactivateNSManagedObjectContextRetrievalFailure()
+	}
 
-//extension FeedStoreChallengeTests: FailableDeleteFeedStoreSpecs {
-//
-//	func test_delete_deliversErrorOnDeletionError() throws {
-////		let sut = try makeSUT()
-////
-////		assertThatDeleteDeliversErrorOnDeletionError(on: sut)
-//	}
-//
-//	func test_delete_hasNoSideEffectsOnDeletionError() throws {
-////		let sut = try makeSUT()
-////
-////		assertThatDeleteHasNoSideEffectsOnDeletionError(on: sut)
-//	}
-//
-//}
+	func test_retrieve_hasNoSideEffectsOnFailure() throws {
+		let sut = try makeSUT()
+
+		activateNSManagedObjectContextRetrievalFailure()
+		
+		assertThatRetrieveHasNoSideEffectsOnFailure(on: sut)
+		
+		deactivateNSManagedObjectContextRetrievalFailure()
+	}
+
+}
+
+extension FeedStoreChallengeTests: FailableInsertFeedStoreSpecs {
+
+	func test_insert_deliversErrorOnInsertionError() throws {
+		let sut = try makeSUT()
+
+		activateNSManagedObjectContextSaveFailure()
+		
+		assertThatInsertDeliversErrorOnInsertionError(on: sut)
+		
+		deactivateNSManagedObjectContextSaveFailure()
+	}
+
+	func test_insert_hasNoSideEffectsOnInsertionError() throws {
+		let sut = try makeSUT()
+
+		activateNSManagedObjectContextSaveFailure()
+		
+		assertThatInsertHasNoSideEffectsOnInsertionError(on: sut)
+		
+		deactivateNSManagedObjectContextSaveFailure()
+	}
+
+}
+
+extension FeedStoreChallengeTests: FailableDeleteFeedStoreSpecs {
+
+	func test_delete_deliversErrorOnDeletionError() throws {
+		let sut = try makeSUT()
+
+		activateNSManagedObjectContextDeleteFailure()
+		
+		assertThatDeleteDeliversErrorOnDeletionError(on: sut)
+		
+		deactivateNSManagedObjectContextDeleteFailure()
+	}
+
+	func test_delete_hasNoSideEffectsOnDeletionError() throws {
+		let sut = try makeSUT()
+		
+		assertThatDeleteHasNoSideEffectsOnDeletionError(on: sut)
+	}
```

Comment:
This test is missing the failure setup:

```suggestion
	func test_delete_hasNoSideEffectsOnDeletionError() throws {
		let sut = try makeSUT()
		
		activateNSManagedObjectContextDeleteFailure()
		
		assertThatDeleteHasNoSideEffectsOnDeletionError(on: sut)
		
		deactivateNSManagedObjectContextDeleteFailure()
	}
```

Reply:
Added failure setup, the previous deletion error simulation helped this test pass. There was a problem with the delete error simulation.